etc/scripts/timetable/import.js

```diff
@@ -94,11 +98,24 @@ GrassHopper.init(config, function(err) {
                     log().info('Creating courses');
                     _createCourses(ctx, courses.slice(), function() {
 
+                        // Take a deep copy of the courses before we import them. The import
+                        // operation might make in-place modifications (such as deleting series)
+                        // which would trip up the borrowing process later
+                        var copyCourses = _.cloneDeep(courses);
                         log().info('Importing the tree under each course');
-                        _importCourses(ctx, courses, function() {
+                        _importCourses(ctx, courses.slice(), function() {
+
+                            log().info('Exporting the tree under each course');
+                            _exportCourses(ctx, copyCourses.slice(), function() {
```

Comment:
It's not entirely clear that the reason we have to export within an import process is to get the new ids until you get deep into the comments of `_exportCourses`. Can you add a line of docs explaining what's going on here?